exchanges/huobi/huobi_test.go

```diff
@@ -1825,30 +1827,27 @@ func TestPairFromContractExpiryCode(t *testing.T) {
 	today := time.Now()
 	today = time.Date(today.Year(), today.Month(), today.Day(), 0, 0, 0, 0, tz) // Do not use Truncate; https://github.com/golang/go/issues/55921
 
-	for _, cType := range contractExpiryNames {
-		p, err := e.pairFromContractExpiryCode(currency.Pair{
-			Base:  currency.BTC,
-			Quote: currency.NewCode(cType),
+	require.NotEmpty(t, e.futureContractCodes, "At least one contract code must be loaded")
+
+	for cType, cachedContract := range e.futureContractCodes {
```

Comment:
`e.futureContractCodes` should have a lock

Reply:
I guess I'll cave on this.
I didn't see the point, because it's a test, and we just instantiated `e`.

Reply:
Fixed https://github.com/gbjk/gocryptotrader/commit/55bfc1d4b

Reply:
Yeah I see your point now. There is a e.futureContractCodes above as well maybe just 

```go
e.futureContractCodesMutex.RLock()
wow := maps.Copy(e.futureContractCodes)
e.futureContractCodesMutex.RUnlock()
```

Or revert just to get rid of this linter issue.